rfcs/framework-0063-notification-system.md

```diff
@@ -0,0 +1,422 @@
+        "community": community,
+        "record": record,
+    }
+)))
+```
+
+#### Notification examples
+
+| Action             | Who triggers   | Who receives                                                                            |
+| ------------------ | -------------- | --------------------------------------------------------------------------------------- |
+| Comment create     | Creating user  | All involved in conversation                                                            |
+| Comment edit       | Editing user   | All involved in conversation                                                            |
+| Comment delete     | Deleting user  | All involved in conversation                                                            |
+|                    |                |                                                                                         |
+| Submission create  | Creating user  | Community owners/managers                                                               |
+| Submission accept  | Accepting user | User who created submission                                                             |
+| Submission decline | Declining user | User who created submission                                                             |
+| Submission edit    | Editing user   | Either user who created submission or Community owners/managers (depends on who edited) |
+|                    |                |                                                                                         |
+| Invitation accept  | Accepting user | User who created invitation                                                             |
+| Invitation decline | Declining user | User who created invitation                                                             |
+| Invitation expire  | System         | User who created invitation                                                             |
+
+## Example
+
+> Show a concrete example of what the RFC implies. This will make the consequences of adopting the RFC much clearer.
+
+> As with other sections, use it if it makes sense for your RFC.
+
+## How we teach this
+
+> What names and terminology work best for these concepts and why? How is this idea best presented? As a continuation of existing Invenio patterns, or as a wholly new one?
+
+> Would the acceptance of this proposal mean the Invenio documentation must be re-organized or altered? Does it change how Invenio is taught to new users at any level?
+
+> How should this feature be introduced and taught to existing Invenio users?
+
+## Alternatives
+
+An alternative is to rely on the implementation of the event bus and register methods. Though it is not known which data will be passed along for the events and there may be some missing data for the notification system.
+
+## Unresolved questions
+
+Tracking success/failure of delivery is not taken care of. If a task gets stuck or dies, there will not be any mention of this.
```

Comment:
silly question, are we use rabbitmq for messages in this case? do notifications get their own queues?

Reply:
Guess: yes. At the moment we have one queue per record type (e.g. funders, awards, names, etc.) so I guess notifs will have their own (or even more fine grained). This would also allow to set priorities to the queues (🤔), useful?

Reply:
For now, there's no real benefit in creating separate queues for the notification message transmission itself. Using separate queues would also go in the direction of events/message bus, which we're trying to avoid to reduce complexity.

The plan is to just use Celery tasks at the moment that get shipped to the primary/default queue. This can be configured of course, but currently in RDM we don't have multiple Celery queues with different priorities, I to reduce deployment/infrastructure complexity.